crates/red_knot_python_semantic/resources/mdtest/type_of/basic.md

```diff
@@ -0,0 +1,89 @@
+# type special form
+
+## Class literal
+
+```py
+class A: ...
+
+def f() -> type[A]:
+    return A
+
+reveal_type(f())  # revealed: type[A]
+```
+
+## Nested class literal
+
+```py
+class A:
+    class B: ...
+
+def f() -> type[A.B]:
+    return A.B
+
+reveal_type(f())  # revealed: type[B]
+```
+
+## Deeply nested class literal
+
+```py
+class A:
+    class B:
+        class C: ...
+
+def f() -> type[A.B.C]:
+    return A.B.C
+
+reveal_type(f())  # revealed: type[C]
+```
+
+## Class literal from another module
+
+```py
+from a import A
+
+def f() -> type[A]:
+    return A
+
+reveal_type(f())  # revealed: type[A]
+```
+
+```py path=a.py
+class A: ...
+```
+
+## Qualified class literal from another module
+
+```py
+import a
+
+def f() -> type[a.B]:
+    return a.B
+
+reveal_type(f())  # revealed: type[B]
+```
+
+```py path=a.py
+class B: ...
+```
+
+## Deeply qualified class literal from another module
+
+```py path=a/test.py
+import a.b
+
+# TODO: no diagnostic
+# error: [unresolved-attribute]
+def f() -> type[a.b.C]:
+    # TODO: no diagnostic
+    # error: [unresolved-attribute]
+    return a.b.C
```

Comment:
I left this as a TODO because I this is not specific to `type[]`.  The `unresolved-attribute` error I get is:

> Type `<module 'System("/src/a/b.py")'>` has no attribute `b`

which suggests that our import logic is binding `a` as `a.b` instead of `a`.  Am I interpreting that right?

Reply:
Yes, it looks like you are! I think this is a previously-not-known bug, unless @AlexWaygood was already aware of it. Want to file an issue (labeled `red-knot`)? Could also be an interesting one for you to look into, would get you some exposure to the module resolver, which I don't think any of the other tasks do.

Reply:
I think I already knew about it, and I _think_ we already have a TODO about it... whether we have an issue is another question... it's late here, so I won't  look into whether we do or don't this evening :-)

Reply:
Opened https://github.com/astral-sh/ruff/issues/14826, can close it as a dup if it turns out we have another one already open